lib/iris/tests/unit/experimental/ugrid/test_MeshCoord.py

```diff
@@ -0,0 +1,394 @@
+        meshcoord = self._create_common_mesh()
+        meshcoord2 = self._create_common_mesh(axis="y")
+        self.assertNotEqual(meshcoord2, meshcoord)
+
+
+class Test__copy(tests.IrisTest):
+    def test_basic(self):
+        meshcoord = _create_test_meshcoord()
+        meshcoord2 = meshcoord.copy()
+        self.assertIsNot(meshcoord2, meshcoord)
+        self.assertEqual(meshcoord2, meshcoord)
+        # In this case, they should share *NOT* copy the Mesh object.
+        self.assertIs(meshcoord2.mesh, meshcoord.mesh)
+
+    def test_fail_copy_newpoints(self):
+        meshcoord = _create_test_meshcoord()
+        with self.assertRaisesRegex(ValueError, "Cannot change the content"):
+            meshcoord.copy(points=meshcoord.points)
+
+    def test_fail_copy_newbounds(self):
+        meshcoord = _create_test_meshcoord()
+        with self.assertRaisesRegex(ValueError, "Cannot change the content"):
+            meshcoord.copy(bounds=meshcoord.bounds)
+
+
+class Test__getitem__(tests.IrisTest):
+    def test_slice_wholeslice_1tuple(self):
+        # The only slicing case that we support, to enable cube slicing.
+        meshcoord = _create_test_meshcoord()
+        meshcoord2 = meshcoord[
+            :,
+        ]
+        self.assertIsNot(meshcoord2, meshcoord)
+        self.assertEqual(meshcoord2, meshcoord)
+        # In this case, we should *NOT* copy the linked Mesh object.
+        self.assertIs(meshcoord2.mesh, meshcoord.mesh)
+
+    def test_slice_whole_slice_singlekey(self):
+        # A slice(None) also fails, if not presented in a 1-tuple.
+        meshcoord = _create_test_meshcoord()
+        with self.assertRaisesRegex(ValueError, "Cannot index"):
+            meshcoord[:]
+
+    def test_fail_slice_part(self):
+        meshcoord = _create_test_meshcoord()
+        with self.assertRaisesRegex(ValueError, "Cannot index"):
+            meshcoord[:1]
+
+
+class Test_cube_containment(tests.IrisTest):
+    # Check that we can put a MeshCoord into a cube, and have it behave just
+    # like a regular AuxCoord.
+    def setUp(self):
+        meshcoord = _create_test_meshcoord()
+        data_shape = (2,) + _TEST_POINTS.shape
+        cube = Cube(np.zeros(data_shape))
+        cube.add_aux_coord(meshcoord, 1)
+        self.meshcoord = meshcoord
+        self.cube = cube
+
+    def test_added_to_cube(self):
+        meshcoord = self.meshcoord
+        cube = self.cube
+        self.assertIn(meshcoord, cube.coords())
+
+    def test_cube_dims(self):
+        meshcoord = self.meshcoord
+        cube = self.cube
+        self.assertEqual(meshcoord.cube_dims(cube), (1,))
+        self.assertEqual(cube.coord_dims(meshcoord), (1,))
+
+    def test_find_by_name(self):
+        meshcoord = self.meshcoord
+        cube = self.cube
+        self.assertIs(cube.coord(standard_name="longitude"), meshcoord)
+        self.assertIs(cube.coord(long_name="long-name"), meshcoord)
+        # self.assertIs(cube.coord(var_name="var"), meshcoord)
```

Comment:
@pp-mo Nuke this? Or was it a reminder to do some additional testing specifically for `var_name`?

Reply:
No, it's dead code left over from a previous approach. 
(!)